From 35a5bea50e26ea0555738e4d5afb7a249e3de67c Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Tue, 12 Sep 2017 02:53:35 +0000
Subject: [PATCH] MORAY-431 Errors after query has timed out are emitted when
 there are no "error" listeners Reviewed by: Jan Wyszynski
 <jan.wyszynski@joyent.com> Reviewed by: David Pacheco <dap@joyent.com>
 Approved by: David Pacheco <dap@joyent.com>

---
 test/sql.test.js | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/test/sql.test.js b/test/sql.test.js
index b53784c..da90e0f 100644
--- a/test/sql.test.js
+++ b/test/sql.test.js
@@ -103,6 +103,24 @@ test('sql - cleanup', function (t) {
     });
 });
 
+test('sql - error after timeout', function (t) {
+    q = c.sql('SELECT pg_sleep(5); SELECT * FROM nonexistent;',
+        [], { timeout: 1 });
+    q.on('error', function (err) {
+        t.ok(VError.hasCauseWithName(err, 'QueryTimeoutError'),
+            'correct error');
+        t.end();
+    });
+    q.on('record', function (r) {
+        t.fail('no row should be returned');
+        t.deepEqual(r, null, 'no row should be returned - value');
+    });
+    q.on('end', function () {
+        t.fail('"error" should have been emitted');
+        t.end();
+    });
+});
+
 test('sql - client timeout respected', function (t) {
     q = c.sql('SELECT pg_sleep(5);', [], { timeout: 1000 });
     q.on('error', function (err) {
@@ -114,7 +132,7 @@ test('sql - client timeout respected', function (t) {
         t.fail('no row should be returned');
         t.deepEqual(r, null, 'no row should be returned - value');
     });
-    q.on('end', function (r) {
+    q.on('end', function () {
         t.fail('"error" should have been emitted');
         t.end();
     });
-- 
2.21.0

